src/ir/item.rs

```diff
@@ -167,12 +199,55 @@ impl Item {
         self.kind().expect_function()
     }
 
-    // This check is needed because even though the type might not contain the
-    // applicable template args itself, they might apply transitively via, for
-    // example, the parent.
-    //
-    // It's kind of unfortunate (in the sense that it's a sort of complex
-    // process, but I think it gets all the cases).
+    /// Checks whether an item contains in its "type signature" some named type.
+    ///
+    /// This function is used to avoid unused template parameter errors in Rust
+    /// when generating typedef declarations, and also to know whether we need
```

Comment:
This is an aside, but I wonder if it would be easier to generate `PhantomData<T>` for all types, even unused types. I *think* this would better match what C++ is actually doing, but it would be a bit uglier to deal with when using the bindings with unused types.